### PR TITLE
#528 Qulice does not enforce any of PMD rules from 'unnecessary' set

### PR DESCRIPTION
**Pull request by HDouss**
Resolving #528 : Adding PMD rule 'unnecessary' and 2 unit tests to ensure PMD is checking against 'unnecessary' set rules.

### CODE REVIEW

**Comment by HDouss** [comment]
@davvd Please assign a CR

**Comment by davvd** [comment]
@hdouss Thanks for your pull request, let me find someone who can review it

**Comment by davvd** [comment]
@pinaf review this one, please

**Comment by HDouss** [comment]
@davvd AppVeyor build [seems to be stuck](https://ci.appveyor.com/project/yegor256/qulice/build/212). Closing and reopening the PR relaunched Travis build, but bot AppVeyor one. Is there a way to relaunch PR checks ?

File: qulice-pmd/src/test/resources/com/qulice/pmd/UnnecessaryFinalModifier.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package foo;
+
+public final class UnnecessaryFinalModifier {
+    
+	public final int foo() {
+	    return 1;
+    }
```
Comment (pinaf) [inline qulice-pmd/src/test/resources/com/qulice/pmd/UnnecessaryFinalModifier.java]: @HDouss extra whitespace here

**Comment by pinaf** [comment]
@HDouss looks good. 1 very minor comment.

**Comment by pinaf** [comment]
@HDouss on appveyor, I believe that a new push might restart it.

**Comment by HDouss** [comment]
@pinaf Thanks ! Done !

**Comment by pinaf** [comment]
@rultor merge

**Comment by HDouss** [comment]
@krzyk Please confirm the merge.

**Comment by krzyk** [comment]
@rultor mege pls

**Comment by HDouss** [comment]
@krzyk Could you please reask the merge ?
PS : there was a typo in your comment 
> mege

**Comment by krzyk** [comment]
@HDouss oops :)

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by HDouss** [comment]
@krzyk Thanks !

**Comment by davvd** [comment]
@rultor deploy

**Comment by davvd** [comment]
@pinaf Many thanks! **19 mins** were added to your account in Transaction ID `AP-3T430411C5511000B` (task took 42 hours and 51 mins)

you're getting extra minutes here (c=4)

+19 added to your rating, at the moment it is: [+4479](http://www.netbout.com/b/35128?open=rating)

**Pull request merged**